src/dmd/root/speller.d

```diff
@@ -43,18 +43,17 @@ private bool combineSpellerResult(T)(ref T p, ref int cost, T np, int ncost)
     return false;
 }
 
-private auto spellerY(alias dg)(const(char)* seed, size_t seedlen, size_t index, ref int cost)
+private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
 {
-    if (!seedlen)
+    if (!seed.length)
         return null;
-    assert(seed[seedlen] == 0);
     char[30] tmp;
-    char* buf;
-    if (seedlen <= tmp.sizeof - 2)
-        buf = tmp.ptr;
+    char[] buf;
+    if (seed.length <= tmp.sizeof - 1)
+        buf = tmp;
     else
     {
-        buf = cast(char*)alloca(seedlen + 2); // leave space for extra char
+        buf = new char[](seed.length + 1); // leave space for extra char
```

Comment:
Why the change from alloca to GC ?

Reply:
Ping on this

Reply:
Sorry about the delay. It was a little late when I had finished that and must've misread the function as malloc or something like that.

Reply:
Although, thinking about it, this alloca would be the only thing keeping this function from being inferred as `@safe` (when the dg parameter is also `@safe`, of course)

Reply:
We're still quite far away from having DMD `@safe`. I think we can leave this for future improvements.